test/unit/initialize.test.js

```diff
@@ -65,4 +65,60 @@ describe('etch.initialize(component)', () => {
       etch.initialize(component)
     }).to.throw(/invalid falsy value/)
   })
+
+  it('allows for multi dimensional refs', async function () {
```

Comment:
Forgive me if I'm missing it, but do you have a test for the removal case. I'm specifically concerned about the logic for removing sub-object when all refs at a given level of nesting disappear.